src/test/java/seedu/clinic/logic/parser/AddCommandParserTest.java

```diff
@@ -64,37 +68,53 @@ public void parse_allFieldsPresent_success() {
         Warehouse expectedWarehouse = new WarehouseBuilder(B).build();
 
         // whitespace only preamble
-        assertParseSuccess(parser, PREAMBLE_WHITESPACE + NAME_DESC_BOB2 + PHONE_DESC_BOB
-                + EMAIL_DESC_BOB + REMARK_DESC_BOB, new AddCommand(expectedSupplier));
-        assertParseSuccess(parser, PREAMBLE_WHITESPACE + NAME_DESC_WAREHOUSE_B2 + PHONE_DESC_WAREHOUSE_B
-                + ADDRESS_DESC_WAREHOUSE_B + REMARK_DESC_BOB, new AddCommand(expectedWarehouse));
+        assertParseSuccess(parser, PREAMBLE_WHITESPACE + VALID_TYPE_SUPPLIER + NAME_DESC_BOB2
+                + PHONE_DESC_BOB + EMAIL_DESC_BOB + REMARK_DESC_BOB, new AddCommand(expectedSupplier));
+        assertParseSuccess(parser, PREAMBLE_WHITESPACE + VALID_TYPE_WAREHOUSE + NAME_DESC_WAREHOUSE_B2
+                + PHONE_DESC_WAREHOUSE_B + ADDRESS_DESC_WAREHOUSE_B + REMARK_DESC_WAREHOUSE_B,
+                new AddCommand(expectedWarehouse));
 
         // multiple names - last name accepted
-        assertParseSuccess(parser, NAME_DESC_AMY2 + NAME_DESC_BOB2 + PHONE_DESC_BOB + EMAIL_DESC_BOB
-                + EMAIL_DESC_BOB, new AddCommand(expectedSupplier));
-        assertParseSuccess(parser, NAME_DESC_WAREHOUSE_A2 + NAME_DESC_WAREHOUSE_B2 + PHONE_DESC_WAREHOUSE_B
-                + ADDRESS_DESC_WAREHOUSE_B + REMARK_DESC_WAREHOUSE_B, new AddCommand(expectedWarehouse));
+        assertParseSuccess(parser, VALID_TYPE_SUPPLIER + NAME_DESC_AMY2 + NAME_DESC_BOB2 + PHONE_DESC_BOB
+                + EMAIL_DESC_BOB + EMAIL_DESC_BOB + REMARK_DESC_BOB, new AddCommand(expectedSupplier));
```

Comment:
There seems to be an extra `EMAIL_DESC_BOB` here!

Reply:
Oh yes, thanks for the good spot!